CtaReactiveMaster/Sources/Home/HomeViewController.swift

```diff
@@ -6,10 +6,112 @@
 //
 
 import UIKit
+import Alamofire
+import SafariServices
 
 final class HomeViewController: UIViewController {
-
+    @IBOutlet private weak var tableView: UITableView!{
+        didSet{
+            tableView.dataSource = self
+            tableView.delegate = self
+            tableView.register(NewsTableViewCell.nib, forCellReuseIdentifier: NewsTableViewCell.identifier)
+            tableView.refreshControl = UIRefreshControl()
+        }
+    }
+    
+    @IBOutlet private weak var activityIndicator: UIActivityIndicatorView!
+    private var articles:[Article] = []
+    
     override func viewDidLoad() {
         super.viewDidLoad()
+        
+        navigationItem.title = "NewsAPI"
+        navigationController?.navigationBar.titleTextAttributes
+            = [NSAttributedString.Key.font: UIFont(name: "Times New Roman", size: 24)!]
+        navigationController?.navigationBar.backgroundColor = UIColor.brown
+        
+        activityIndicator.hidesWhenStopped = true
+        tableView.refreshControl?.addTarget(self, action: #selector(refresh(sender:)), for: .valueChanged)
+       
+        activityIndicator.startAnimating()
+        fetchNewsAPI()
+    }
+    
+    private func fetchNewsAPI(){
+        
+        AF.request("http://newsapi.org/v2/everything?q=bitcoin&from=2020-12-14&sortBy=publishedAt&apiKey=67945148525042b9b63954def7a50c38").response { [weak self] response in
+        
+            guard let self = self else {return}
+
+            switch response.result {
+                case .success(let data):
+                    guard let data = data else{ return }
+                    
+                    do{
+                        //APIのデータに従った構造体を書く → 取得するJSONのデータに合わせた構造体jsonDataを定義
+                        let jsonDecoder = JSONDecoder()
+                        jsonDecoder.dateDecodingStrategy = .iso8601
+                        let news = try jsonDecoder.decode(News.self, from: data)
+                        self.articles = news.articles
+                    }
+                    catch let error{
+                        print(error)
+                        self.showRetryAlert(with: error, retryhandler: self.fetchNewsAPI)
+                        self.AfterFetch()
+                        return
+                    }
+                    self.tableView.reloadData()
+                    
+                case .failure(let error):
+                    //selfをあらかじめアンラップすることによりfetchnewsAPI → ?? () -> ()を使用することがない状態に
+                    self.showRetryAlert(with: error, retryhandler: self.fetchNewsAPI)
+                    print(error)
+                    
+                    
+
+            }
+            self.AfterFetch()
+        }
+    }
+    
+    private func AfterFetch(){
```

Comment:
関数は基本 lower camel で記述してほしい
```suggestion
    private func afterFetch(){
```